docs/docs/how_to.md

```diff
@@ -0,0 +1,227 @@
+going to continue with our OOP approach.
+
+```python
+def __init__(self):
+    self.robot = Robot()
+    self.name = self.robot.getName()
+    ...
+```
+
+The `__init__` something like constructor of the class and is called when the object
+is created. We initialize some important variables there. The most important one
+is the `Robot` instance, which allows us to get access to the so called devices
+like motor (for controlling the speed) or receiver (for reading data from supervisor).
+**The name and the team of your robot** can be determined by calling `self.robot.getName()`.
+It will give you one of `{"B1", "B2", "B3", "Y1", "Y2", "Y3"}`. The first letter determines
+the team ("Blue", "Yellow"), while the second one is the robot's identifier.
+
+```python
+def get_new_data(self):
+    ...
+``` 
+
+We are not going to explain this deeply. This function is to parse the incoming
+data from supervisor. Feel free to copy and use it. The resulting dictionary
+contains positions of all of the robots as well as the position of the ball.
+
+```python
+def run(self):
+```
+
+This is the method which contains the logic for controlling the robot.
+
+```python
+while self.robot.step(TIME_STEP) != -1:
+```
+
+**The `step` function is crucial and must be used in every controller.
+This function synchronizes the sensor and actuator data between Webots and the controllers.**
+
+
+```python
+if self.receiver.getQueueLength() > 0:
+```
+
+Before reading data, its important to check if there is actually something
+in the queue.
+
+```python
+data = self.get_new_data()
+
+robot_pos = data[self.name]
+ball_pos = data['ball']
+
+self.left_motor.setVelocity(1)
+self.right_motor.setVelocity(-1)
+```
+
+And finally, read the new data received from supervisor, do some calculations
+and set the speed of the motors.
+
+```python
+my_robot = MyRobot()
+my_robot.run()
+```
+
+In the very end of our program, we initialize the robot and call the method `run`
+in order to run the code.
+
+
+## Importing shared code
+
+Each team consists out of three robots. These robots might share some of the code
+and its actually a good practice not to duplicate code all over the place.
+
+Imagine you have the folder structure which looks like this
+
+```
+controllers/
+├── robot1/
+│   └── robot1.py
+|   └── utils.py
+├── robot2/
+│   └── robot2.py
+├── robot3/
+│   └── robot3.py
+```
+
+and within `robot2.py` you want to import some useful code from the `utils.py`
+located in `robot1` folder. This is possible, but you need to put the code below
+on top of your `robot2.py` file.
+
+```python
+import sys
+from pathlib import Path
+sys.path.append(str(Path('.').absolute().parent))
+from robot1 import utils
+``` 
+
+This will ensure that the code from `robot1` is importable (adds the parent folder
+to the PATH)
+
+If you want to import `utils` within the `robot1`, you do no have to add these
+magic lines to `robot1.py`, but instead just call
+
+```python
+import utils
+```
```

Comment:
@Adman it would be great to add a bit of a background so as to why is that the case

Reply:
Added.